docs/items/items.md

```diff
@@ -30,4 +30,29 @@ Registering an Item
 
 Items must be [registered][registering] to function.
 
-[registering]: ../concepts/registries.md#registering-things
+Coloring an Item
+----------------
+
+There are items, textures of which are colored within the code, and in fact, there are a lot of such items. For example, leather helmets, spawn eggs, potions and other.
+
+### Item Color Handlers
+
+An item color handler is required to color an item and is an instance of `IItemColor`, and therefore implements the following method.
+
+```java
+int getColorFromItemstack(
+  ItemStack stack, 
+  int tintIndex)
+```
+
+This method returns a hex representation of a color in an integer.
+
+If the subject of coloring is a regular `Item`, the tint index will show an index of item's texture layer that is being drawn. Layers of item's texture are specified in its model JSON file.
```

Comment:
this is only true if you use `builtin/generated`. you can have a custom model with your tint indices, and have a custom IItemColor that looks for them.

I would reword this section to say that the tint index specified in the JSON is passed to getColorFromItemstack, but for `builtin/generated` models, each layer automatically gets a tint index assigned.